playground/backend/cmd/server/controller.go

```diff
@@ -155,207 +180,11 @@ func (controller *playgroundController) GetExampleOutput(ctx context.Context, in
-
-		// set to cache pipelineId: cache.SubKey_Status: pb.Status_STATUS_VALIDATION_ERROR
-		setToCache(ctx, cacheService, pipelineId, cache.Status, pb.Status_STATUS_VALIDATION_ERROR)
-	case pb.Status_STATUS_COMPILE_ERROR:
-		logger.Errorf("%s: Compile: err: %s, output: %s\n", pipelineId, err.Error(), data)
-
-		// set to cache pipelineId: cache.SubKey_CompileOutput: err.Error()
-		setToCache(ctx, cacheService, pipelineId, cache.CompileOutput, "error: "+err.Error()+", output: "+string(data))
-
-		// set to cache pipelineId: cache.SubKey_Status: pb.Status_STATUS_ERROR
-		setToCache(ctx, cacheService, pipelineId, cache.Status, pb.Status_STATUS_COMPILE_ERROR)
-	case pb.Status_STATUS_ERROR:
-		logger.Errorf("%s: Run: err: %s, output: %s\n", pipelineId, err.Error(), data)
-
-		// set to cache pipelineId: cache.SubKey_RunOutput: err.Error()
-		setToCache(ctx, cacheService, pipelineId, cache.RunOutput, "error: "+err.Error()+", output: "+string(data))
-
-		// set to cache pipelineId: cache.SubKey_Status: pb.Status_STATUS_ERROR
-		setToCache(ctx, cacheService, pipelineId, cache.Status, pb.Status_STATUS_ERROR)
-	}
-}
-
-// processSuccess processes case after successful code processing via setting a corresponding status and output to cache
-func processSuccess(ctx context.Context, output []byte, pipelineId uuid.UUID, cacheService cache.Cache, status pb.Status) {
-	switch status {
-	case pb.Status_STATUS_COMPILING:
-		logger.Infof("%s: Validate() finish\n", pipelineId)
-
-		// set to cache pipelineId: cache.SubKey_Status: pb.Status_STATUS_EXECUTING
-		setToCache(ctx, cacheService, pipelineId, cache.Status, pb.Status_STATUS_COMPILING)
-	case pb.Status_STATUS_EXECUTING:
-		logger.Infof("%s: Compile() finish\n", pipelineId)
-
-		// set to cache pipelineId: cache.SubKey_CompileOutput: output
-		setToCache(ctx, cacheService, pipelineId, cache.CompileOutput, string(output))
-
-		// set to cache pipelineId: cache.SubKey_Status: pb.Status_STATUS_EXECUTING
-		setToCache(ctx, cacheService, pipelineId, cache.Status, pb.Status_STATUS_EXECUTING)
-	case pb.Status_STATUS_FINISHED:
-		logger.Infof("%s: Run() finish\n", pipelineId)
-
-		// set to cache pipelineId: cache.SubKey_RunOutput: output
-		setToCache(ctx, cacheService, pipelineId, cache.RunOutput, string(output))
-
-		// set to cache pipelineId: cache.SubKey_Status: pb.Status_STATUS_FINISHED
-		setToCache(ctx, cacheService, pipelineId, cache.Status, pb.Status_STATUS_FINISHED)
-	}
-}
-
 // setToCache puts value to cache by key and subKey
-func setToCache(ctx context.Context, cacheService cache.Cache, key uuid.UUID, subKey cache.SubKey, value interface{}) {
-	if err := cacheService.SetValue(ctx, key, subKey, value); err != nil {
+func setToCache(ctx context.Context, cacheService cache.Cache, key uuid.UUID, subKey cache.SubKey, value interface{}) error {
+	err := cacheService.SetValue(ctx, key, subKey, value)
+	if err != nil {
 		logger.Errorf("%s: cache.SetValue: %s\n", key, err.Error())
 	}
+	return err
```

Comment:
What do you think about moving such functions to some `utils` package? This one, for example, could lay in `utils/cache_utils.go`?

Reply:
Moved.